packages/network/src/v2/schemas/decodeDynamicField.ts

```diff
@@ -0,0 +1,18 @@
+import { SchemaType } from "@latticexyz/schema-type";
+
+const unsupportedDynamicField = (fieldType: SchemaType): never => {
+  throw new Error(`Unsupported dynamic field type: ${SchemaType[fieldType] ?? fieldType}`);
+};
+
+// TODO: decode per dynamic field type, so we get better return types for each field type
```

Comment:
Could we create a type like the v1 `ContractSchemaValueTypes` and make the function generic? (`decodeDynamicField<T extends SchemaType>(fieldType: T, data: ArrayBuffer): SchemaTypeToTsType<T>`)